AndroidHomeApplication/app/src/main/java/com/example/androidhomeapplication/reduxPagination/Store.kt

```diff
@@ -0,0 +1,103 @@
+package com.example.androidhomeapplication.reduxPagination
+
+class Store<T> {
+    private var state: State = State.Empty
+
+    var render: (State) -> Unit = {}
+    var sideEffects: (SideEffect) -> Unit = {}
+
+    fun proceed(action: Action) {
+        val newState = reducer<T>(action, state) { sideEffect ->
+            sideEffects(sideEffect) // вынести на другой поток
+        }
+        if (newState != state) {
+            state = newState
+            render(state)
+        }
+    }
+
+    private fun <T> reducer(
+        action: Action,
+        state: State,
+        sideEffectListener: (SideEffect) -> Unit
```

Comment:
not idiomatic way
why u have listener for side effect but not return type for new state?
also it's not clear can be few side effects or nor?
just return pair of side effect and newstate